operator/pkg/controllers/etcd/service.go

```diff
@@ -0,0 +1,70 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package etcd
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/awslabs/kit/operator/pkg/apis/infrastructure/v1alpha1"
+	"github.com/awslabs/kit/operator/pkg/utils/object"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+func (c *Controller) reconcileService(ctx context.Context, controlPlane *v1alpha1.ControlPlane) error {
+	return c.kubeClient.Ensure(ctx, object.WithOwner(controlPlane, &v1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      serviceNameFor(controlPlane.ClusterName()),
+			Namespace: controlPlane.NamespaceName(),
+			Labels:    labelFor(controlPlane.ClusterName()),
+		},
+		Spec: v1.ServiceSpec{
+			ClusterIP: v1.ClusterIPNone,
```

Comment:
ClusterIPNone seems wrong here. Can we leave this empty?

Reply:
Need to explicitly set this to `None`, to make this a [headless service](https://kubernetes.io/docs/concepts/services-networking/service/#headless-services).